compiler/rustc_codegen_llvm/src/debuginfo/metadata/enums/cpp_like.rs

```diff
@@ -370,6 +681,21 @@ fn build_union_fields_for_direct_tag_generator<'ll, 'tcx>(
     let common_upvar_names = closure_saved_names_of_captured_variables(cx.tcx, generator_def_id);
     let variant_range = generator_substs.variant_range(generator_def_id, cx.tcx);
 
+    let tag_base_type = tag_base_type(cx, generator_type_and_layout);
+
+    let variant_names_type_di_node = build_variant_names_type_di_node(
+        cx,
+        generator_type_di_node,
+        variant_range
+            .clone()
+            .map(|variant_index| (variant_index, GeneratorSubsts::variant_name(variant_index))),
+    );
+
+    let discriminants: FxHashMap<VariantIdx, DiscrResult> = generator_substs
```

Comment:
nit: `IndexVec` might be a better choice here

Reply:
Fixed.